fcrepo-message-consumer-core/src/main/java/org/fcrepo/indexer/NamedFieldsRetriever.java

```diff
@@ -86,37 +91,55 @@ public NamedFields get() {
         try {
             final Model rdf = rdfr.get();
             if (!rdf.contains(createResource(uri.toString()), INDEXING_TRANSFORM_PREDICATE)) {
-                LOGGER.info(
-                        "Found no property locating LDPath transform for: {}, will not retrieve transformed content.",
-                        uri);
-                throw new AbsentTransformPropertyException(uri.toString());
+                LOGGER.info("Looking up property locating LDPath transform for: {}", uri);
+                // make an initial HEAD request and check Link headers for descriptions located elsewhere
+                final HttpHead headRequest = new HttpHead(uri);
+                final HttpResponse headResponse = httpClient.execute(headRequest);
+                URI descriptionURI = null;
+                final Header[] links = headResponse.getHeaders("Link");
+                if ( links != null ) {
+                    for ( Header h : headResponse.getHeaders("Link") ) {
+                        final Link link = Link.valueOf(h.getValue());
+                        if (link.getRel().equals("describedby")) {
+                            descriptionURI = link.getUri();
+                            LOGGER.debug("Using URI from Link header: {}", descriptionURI);
+                        }
+                    }
+                }
+                if (descriptionURI == null) {
+                    throw new AbsentTransformPropertyException("Property lookup failed");
+                }
+                return getNamedFields(rdf, descriptionURI);
             }
-            final RDFNode indexingTransform =
+            return getNamedFields(rdf, uri);
+        } catch (IOException | HttpException e) {
+            throw propagate(e);
+        }
+    }
+
+    private NamedFields getNamedFields(final Model rdf, final URI uri) throws IOException, HttpException {
```

Comment:
You are on the right track with the PR, but before blindly returning `getNamedFields` above, you should ensure that the triple exists, along the lines of:
`if (!rdf.contains(createResource(uri.toString()), INDEXING_TRANSFORM_PREDICATE))`

That check can be in the `getNamedFields` method, or in the `get` method. In any case, you should check if the triple exists in the supplied rdf, if not, then check again with the `descriptionURI`.

Your current logic checks if the transform triple exists, if not, then it blindly hopes that the triple exists with the `descriptionURI`.